zcash_client_sqlite/src/wallet/init/migrations/add_transaction_views.rs

```diff
@@ -0,0 +1,356 @@
+//! Functions for initializing the various databases.
+use rusqlite::{self, types::ToSql, NO_PARAMS};
+use schemer::{self};
+use schemer_rusqlite::RusqliteMigration;
+
+use std::collections::HashSet;
+
+use uuid::Uuid;
+
+use zcash_primitives::{
+    consensus::{self, BlockHeight, BranchId},
+    transaction::{components::amount::Amount, Transaction},
+};
+
+use super::super::{WalletMigration2, WalletMigrationError};
+
+pub(crate) fn migration_id() -> Uuid {
+    Uuid::parse_str("282fad2e-8372-4ca0-8bed-71821320909f").unwrap()
+}
+
+pub(crate) struct Migration<P> {
+    pub(super) params: P,
+}
+
+impl<P> schemer::Migration for Migration<P> {
+    fn id(&self) -> Uuid {
+        migration_id()
+    }
+
+    fn dependencies(&self) -> HashSet<Uuid> {
+        let mut deps = HashSet::new();
+        deps.insert(WalletMigration2::<P>::id());
+        deps
+    }
+
+    fn description(&self) -> &'static str {
+        "Add transaction summary views & add fee information to transactions."
+    }
+}
+
+impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
+    type Error = WalletMigrationError;
+
+    fn up(&self, transaction: &rusqlite::Transaction) -> Result<(), WalletMigrationError> {
+        transaction.execute_batch("ALTER TABLE transactions ADD COLUMN fee INTEGER;")?;
+
+        let mut stmt_list_txs =
+            transaction.prepare("SELECT id_tx, raw, block FROM transactions")?;
+
+        let mut stmt_set_fee =
+            transaction.prepare("UPDATE transactions SET fee = ? WHERE id_tx = ?")?;
+
+        let mut stmt_find_utxo_value = transaction
+            .prepare("SELECT value_zat FROM utxos WHERE prevout_txid = ? AND prevout_idx = ?")?;
+
+        let mut tx_rows = stmt_list_txs.query(NO_PARAMS)?;
+        while let Some(row) = tx_rows.next()? {
+            let id_tx: i64 = row.get(0)?;
+            let tx_bytes: Option<Vec<u8>> = row.get(1)?;
+            let h: u32 = row.get(2)?;
+            let block_height = BlockHeight::from(h);
+
+            // If only transaction metadata has been stored, and not transaction data, the fee
+            // information will eventually be set when the full transaction data is inserted.
+            if let Some(b) = tx_bytes {
+                let tx =
+                    Transaction::read(&b[..], BranchId::for_height(&self.params, block_height))
+                        .map_err(|e| {
+                            WalletMigrationError::CorruptedData(format!(
+                                "Parsing failed for transaction {:?}: {:?}",
+                                id_tx, e
+                            ))
+                        })?;
+
+                let fee_paid = tx.fee_paid(|op| {
+                    stmt_find_utxo_value
+                        .query_row(&[op.hash().to_sql()?, op.n().to_sql()?], |row| {
+                            row.get(0).map(|i| Amount::from_i64(i).unwrap())
```

Comment:
This unwrap _should_ be fine (it won't trigger for any data that's been inserted by `zcash_client_sqlite`), but there's technically a crash chance here. Open an issue to fix this if you don't want to do so in this PR.